rfcs/12-primitive-shapes.md

```diff
@@ -0,0 +1,538 @@
+
+There is significant complexity in the way seemingly equivalent shapes are defined depending on how they are used. By considering these use cases *first*, the goal is that geometric data structures can be composed depending on the needed functionality, with traits used to ensure the right data structure is used for the right task.
+
+## User-Facing Explanation
+
+Geometric primitives are lightweight representations of geometry that describe the type of geometry as well as its dimensions. These primitives are *not* meshes, but the underlying precise mathematical definition. For example, a circle is:
+
+```rust
+pub struct Circle2d {
+  radius: f32,
+}
+```
+
+Note that a `Circle2d` does not contain any information about the position of the circle. This is due to how shapes are composed to add more complex functionality. Consider the following common use cases of primitives shapes, as well as the the information (Translation and Rotation) that are needed to fully define the shapes for these cases:
+
+| Shape     | Mesh | Bounding | Collision |
+|---        |---|---|---|
+| Sphere    | ✔ | ✔  + Trans | ✔ + Trans |
+| Box    | ✔ | ✔ + Trans (AABB) | ✔ + Trans + Rot (OBB) |
+| Capsule   | ✔ | ❌ | ✔ + Trans + Rot |
+| Cylinder  | ✔ | ❌ | ✔ + Trans + Rot |
+| Cone      | ✔ | ❌ | ✔ + Trans + Rot |
+| Wedge     | ✔ | ❌ | ✔ + Trans + Rot |
+| Plane     | ✔ | ❌ | ✔ |
+| Torus     | ✔ | ❌ | ❌ |
+
+### Bounding vs. Collision
+
+The difference between the two is somewhat semantic. Both bounding and collision check for intersections between bounding volumes or areas. However, bounding volumes are generally used for broad phase checks due to their speed and small size, and are consequently often used in tree structures for spacial queries. Note that both bounding types are not oriented - this reduces the number of comparisons needed to check for intersections during broad phase tests. 
+
+Although the traits for these may look similar, by making them distinct it will better conform to industry terminology and help guide users away from creating performance problems. For example, because a torus is a volume, it is conceivably possible to implement the `Bounding` trait, however it would be a terrible idea to be encouraging users to build BVTs/BVHs out of torii. This is also why the Oriented Bounding Box (OBB) is a collider, and not `Bounding`. Bounding spheres or AABBs are generally always preferred for broad phase bounding checks, despite the OBB's name. For advanced users who want to implement `Bounding` for OBBs, they always have the option of wrapping the OBB in a newtype and implementing the trait.
```

Comment:
I think the bounding and collider section is a bit distracting - the details are entirely out of scope. They are presented to show how these types will be used, and as justification separating transforms from the primitive shapes.

> Maybe instead of not having them, you could achieve more or less the same by not including them in the prelude and having them in a module with a name that indicates their more advanced applications?

If someone wants to make a bounding/collision plugin or an RFC for the engine, then I think your suggestion is probably the way to go, however I think it's simply out of scope.

I prefaced the statement with

> Consider the following common use cases of primitives shapes [...]

and the trait section with

> These traits are provided as reference to illustrate how these might be used. The details of implementation and interface should be determined in a separate RFC, PR, or independent prototypes.

However, bounding and collision seem to be a sticking point in discussion here. Is there something I can do to make it more clear that these details are out of scope?